acls_types.go

```diff
@@ -100,3 +108,28 @@ func (policy ACLPolicy) IsZero() bool {
 
 	return false
 }
+
+// Returns the list of autoApproving namespaces, groups or tags for a given IPPrefix.
+func (autoApprovers *AutoApprovers) GetRouteApprovers(
+	prefix netip.Prefix,
+) ([]string, error) {
+	if prefix.Bits() == 0 {
+		return autoApprovers.ExitNode, nil // 0.0.0.0/0, ::/0 or equivalent
+	}
+
+	approverAliases := []string{}
+
+	for autoApprovedPrefix, autoApproverAliases := range autoApprovers.Routes {
+		autoApprovedPrefix, err := netip.ParsePrefix(autoApprovedPrefix)
+		if err != nil {
+			return nil, err
+		}
+
+		if autoApprovedPrefix.Bits() >= prefix.Bits() &&
```

Comment:
I think this logic might be the wrong way around?

i.e if I have an auto approved prefix of `10.0.0.0/8` and then I try to advertise `10.1.0.0/24`, I would expect `autoApprovedPrefix.Bits()` == 8 and `prefix.Bits()` == 24, or am I misunderstanding this?

I ask because I tried out the parent prefix thing and it doesn't seem to work for me. For now I'm going to add all 255 /24s I need to my ACL to move on xD

Reply:
Yeah you're right, bugger but thanks for the find.

@juanfont @kradalby I'll write and submit a PR to fix this with a regression test

Reply:
Wow, thanks for the speedy response. And also thanks for shipping this very hype feature :)

Reply:
np, nothing good on tv right now 😂 should be fixed in https://github.com/juanfont/headscale/pull/862 hopefully